components/menus/sniper.ts

```diff
@@ -109,19 +114,12 @@ export function createSniperLoadouts(
                     ],
                     LimitedLoadoutUnlockLevel: 0 as number | undefined,
                 },
-                // TODO(Anthony): Use the function to get these details.
-                CompletionData: {
-                    Level: 20,
-                    MaxLevel: 20,
-                    XP: 0,
-                    Completion: 1,
-                    XpLeft: 0,
-                    Id: index,
-                    SubLocationId: "",
-                    HideProgression: false,
-                    IsLocationProgression: false,
-                    Name: index,
-                } as CompletionData,
+                CompletionData: controller.masteryService.getFirearmCompletion(
```

Comment:
I think the only problem with this is that it's not keeping in mind players that just want everything unlocked from the start?

Reply:
For now everything **is** unlocked from the start. As for later when progression is added, I think it would be the same as opting out of normal progression. No matter which method you end up using, I will follow suit.